### PR TITLE
Different settle transition depending on momentum

### PR DESCRIPTION
**Pull request by DouglasWilcox**
The settle animation that transitions from walk/run to idle is too big for very small adjustments, such as nudging the WASD keys just to slightly reposition.  This creates a bool that is true when speed rises above 2.2 m/s.  If true, play the larger settle.

task:  https://highfidelity.atlassian.net/browse/DEV-2324

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Can someone from High Fidelity approve this PR?

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22387/HighFidelity-Beta-PR16467-73202b9.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22387/HighFidelity-Quest-Beta-PR16467-73202b9.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22387/HighFidelity-Beta-PR16467-73202b9.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22387/HighFidelity-Beta-Interface-PR16467-73202b9.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22387/HighFidelity-Beta-PR16467-73202b9.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22387/HighFidelity-Beta-Interface-PR16467-73202b9.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22393/HighFidelity-Beta-PR16467-8e11d91.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22393/HighFidelity-Beta-Interface-PR16467-8e11d91.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22393/HighFidelity-Beta-PR16467-8e11d91.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22393/HighFidelity-Quest-Beta-PR16467-8e11d91.apk)

**Comment by DouglasWilcox** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22397/HighFidelity-Beta-PR16467-8e11d91.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22397/HighFidelity-Quest-Beta-PR16467-8e11d91.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22397/HighFidelity-Beta-PR16467-8e11d91.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22397/HighFidelity-Beta-Interface-PR16467-8e11d91.dmg

**Comment by DouglasWilcox** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22398/HighFidelity-Beta-PR16467-8e11d91.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22398/HighFidelity-Beta-Interface-PR16467-8e11d91.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22398/HighFidelity-Beta-PR16467-8e11d91.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22398/HighFidelity-Beta-Interface-PR16467-8e11d91.exe

File: libraries/animation/src/Rig.cpp [hunk 1]
```diff
@@ -1437,17 +1438,35 @@ void Rig::computeMotionAnimationState(float deltaTime, const glm::vec3& worldPos
...
-                _animVars.set("isNotInputSlow", true);
+                _animVars.set("isNotInput", true); 
+
+                if (_isMovingWithMomentum) {
+                    _animVars.set("isNotInputSlow", true);
+                    _animVars.set("isNotInputNoMomentum", false);
+                } else {
+                    _animVars.set("isNotInputSlow", false);
+                    _animVars.set("isNotInputNoMomentum", true);
+                }
```
Comment (hyperlogic) [inline libraries/animation/src/Rig.cpp]: You can simplify this as:

    _animVars.set("isNotInputSlow", _isMovingWithMomentum);
    _animVars.set("isNotInputNoMomentum", !_isMovingWithMomentum);

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22420/HighFidelity-Beta-PR16467-02045a9.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22420/HighFidelity-Beta-Interface-PR16467-02045a9.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22420/HighFidelity-Beta-PR16467-02045a9.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22420/HighFidelity-Quest-Beta-PR16467-02045a9.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22420/HighFidelity-Beta-PR16467-02045a9.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22420/HighFidelity-Beta-Interface-PR16467-02045a9.exe

**Comment by DouglasWilcox** [comment]
TEST NOTES:
This changes the settle, which is our transition from locomotion -> idle.  The problem I'm trying to solve is the repetition of that one transition, and also the fact that it was too big and bouncy for very small movement adjustments.   

This PR plays a normal settle if your movement goes above 2.2 m/s, small settle if you don't.  It also adds random variants of the normal settle.  

To test, try starting/stopping different directions and different speeds.  It's worth looking at HMD too, but this should not be significantly different in compared to the old version.

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22425/HighFidelity-Beta-PR16467-055d5cd.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22425/HighFidelity-Quest-Beta-PR16467-055d5cd.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22425/HighFidelity-Beta-PR16467-055d5cd.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22425/HighFidelity-Beta-Interface-PR16467-055d5cd.dmg

**Comment by DouglasWilcox** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22427/HighFidelity-Beta-PR16467-055d5cd.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22427/HighFidelity-Beta-Interface-PR16467-055d5cd.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22427/HighFidelity-Beta-PR16467-055d5cd.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22427/HighFidelity-Quest-Beta-PR16467-055d5cd.apk)

**Comment by DouglasWilcox** [comment]
Hold off on merging this - found a bug going into sit

**Comment by DouglasWilcox** [comment]
False alarm - this looks good.  I have a bug in another branch that I wanted to make sure is not in this. 
 Removed the DONOTMERGE tag.

**Comment by DouglasWilcox** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22431/HighFidelity-Beta-PR16467-055d5cd.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22431/HighFidelity-Beta-Interface-PR16467-055d5cd.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22431/HighFidelity-Beta-PR16467-055d5cd.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22431/HighFidelity-Quest-Beta-PR16467-055d5cd.apk)

**Comment by DouglasWilcox** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22453/HighFidelity-Beta-PR16467-055d5cd.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22453/HighFidelity-Beta-Interface-PR16467-055d5cd.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22453/HighFidelity-Beta-PR16467-055d5cd.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22453/HighFidelity-Quest-Beta-PR16467-055d5cd.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22453/HighFidelity-Beta-PR16467-055d5cd.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22453/HighFidelity-Beta-Interface-PR16467-055d5cd.exe

**Pull request merged**